tests/cephfs/client_mds_restriction.py

```diff
@@ -0,0 +1,137 @@
+        mon_node_ip = ",".join(mon_node_ip)
+        mount_points = []
+        fs1 = "cephfs"
+        fs2 = "cephfs-ec"
+        log.info(f"Creating client authorized to {fs1}")
+        fs_util.fs_client_authorize(client1, fs1, "client1", "/", "rw")
+        log.info(f"Creating client authorized to {fs2}")
+        fs_util.fs_client_authorize(client1, fs2, "client2", "/", "rw")
+        kernel_mount_dir = "/mnt/kernel" + "".join(
+            secrets.choice(string.ascii_uppercase + string.digits) for i in range(5)
+        )
+        fuse_mount_dir = "/mnt/fuse" + "".join(
+            secrets.choice(string.ascii_lowercase + string.digits) for i in range(5)
+        )
+        mount_points.extend([kernel_mount_dir, fuse_mount_dir])
+        log.info(f"Mounting {fs1} with client1")
+        fs_util.kernel_mount(
+            clients,
+            kernel_mount_dir,
+            mon_node_ip,
+            new_client_hostname="client1",
+            extra_params=f",fs={fs1}",
+        )
+        fs_util.fuse_mount(
+            clients,
+            fuse_mount_dir,
+            new_client_hostname="client1",
+            extra_params=f" --client_fs {fs1}",
+        )
+        log.info(f"Verifying mount failure for client1 for {fs2}")
+        rc = verify_mount_failure_on_root(
+            fs_util,
+            clients,
+            kernel_mount_dir + "_dir1",
+            fuse_mount_dir + "_dir1",
+            "client1",
+            mon_node_ip,
+            fs_name=f"{fs2}",
+        )
+        if rc == 1:
+            log.error(f"Mount success on {fs2} with client1")
+            return 1
+        kernel_mount_dir = "/mnt/kernel" + "".join(
+            secrets.choice(string.ascii_uppercase + string.digits) for i in range(5)
+        )
+        fuse_mount_dir = "/mnt/fuse" + "".join(
+            secrets.choice(string.ascii_lowercase + string.digits) for i in range(5)
+        )
+        mount_points.extend([kernel_mount_dir, fuse_mount_dir])
+        log.info(f"Mounting {fs2} with client2")
+        fs_util.kernel_mount(
+            clients,
+            kernel_mount_dir,
+            mon_node_ip,
+            new_client_hostname="client2",
+            extra_params=f",fs={fs2}",
+        )
+        fs_util.fuse_mount(
+            clients,
+            fuse_mount_dir,
+            new_client_hostname="client2",
+            extra_params=f" --client_fs {fs2}",
+        )
+        rc = verify_mount_failure_on_root(
+            fs_util,
+            clients,
+            kernel_mount_dir + "_dir2",
+            fuse_mount_dir + "_dir2",
+            "client2",
+            mon_node_ip,
+            fs_name=f"{fs1}",
+        )
+        log.info(f"Verifying mount failure for client2 for {fs1}")
+        if rc == 1:
+            log.error(f"Mount success on {fs1} with client2")
+            return 1
+        return 0
+
+    except CommandFailed as e:
+        log.info(e)
+        log.info(traceback.format_exc())
+        return 1
```

Comment:
Can we remove the commandFailed except block.
As we are covering same in generic except block